eth-rpc/client.nim

```diff
@@ -72,32 +79,38 @@ proc connect*(self: RpcClient, address: string, port: Port): Future[void]
 proc processData(self: RpcClient) {.async.} =
   while true:
     # read until no data
-    let line = await self.socket.recvLine()
-
+    # TODO: we need to limit number of bytes we going to read, without any
+    # limits server can fill all memory it can here.
+    let line = await self.transp.readLine()
     if line == "":
       # transmission ends
-      self.socket.close()  # TODO: Do we need to drop/reacquire sockets?
-      self.socket = newAsyncSocket()
+      self.transp.close()
       break
-    
+
     processMessage(self, line)
   # async loop reconnection and waiting
-  await connect(self, self.address, self.port)
+  self.transp = await connect(self.address)
 
 proc connect*(self: RpcClient, address: string, port: Port) {.async.} =
-  await self.socket.connect(address, port)
-  self.address = address
-  self.port = port
+  # TODO: `address` hostname can be resolved to many IP addresses, we are using
+  # first one, but maybe it would be better to iterate over all IP addresses
+  # and try to establish connection until it will not be established.
+  let addresses = resolveTAddress(address & ":" & $int(port))
```

Comment:
Why don't we add more construction options for the Address type (i.e. taking an address and a port as separate parameters)? These unnecessary string concatenations keep popping up.

Reply:
Yep, i will fix it, after fix Nim.